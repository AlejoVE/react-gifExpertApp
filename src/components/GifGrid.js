import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
	const { data: images, loading } = useFetchGifs(category);

	return (
		<>
			<h3 className="animate__animated animate__fadeIn">{category}</h3>
			<div className="card-grid">
				{/* este simbolo && significa que si está cargando, ponga un párrago que diga loading
				Si no está cargando, no hace nada */}
				{loading && <p className="animate__animated animate__flash">Loading...</p>}
				{
					// prettier/ignore
					images.map((img) => (
						<GifGridItem {...img} key={img.id} />
					))
				}
			</div>
		</>
	);
};

GifGrid.propTypes = {
	category : PropTypes.string.isRequired,
}