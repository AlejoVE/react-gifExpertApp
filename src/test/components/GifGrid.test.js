import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom'

//con esto fingiremos la llamada a useFetchGifs y controlaremos la informacion que eso va a responder
jest.mock('../../hooks/useFetchGifs');


describe('tests in <GifGrid/> component', () => {
	const category = 'Bikes'
	
	test('Should show <GifGrid/> correctly ', () => {

		useFetchGifs.mockReturnValue({
			data: [],
			loading: true
		});
		
		let wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
	});
	
	test('should  show items when images are loaded with useFetchGifs', () => {
		const gifs = [{
			id: 'ABC',
			title: 'anything',
			url: 'https://localhost/myimage.jpg'
		}];
		
		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false
		});
		
		let wrapper = shallow(<GifGrid category={category} />);

		expect(wrapper.find('p').exists()).toBe(false);
		
		//esta es una prueba m√°s precisa. Se usa para determinar si el componente tiene la cantidad de elemtnos
		//que se esperan
		expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
	})
	
});
