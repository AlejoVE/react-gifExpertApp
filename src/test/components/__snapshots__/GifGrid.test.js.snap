// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <GifGrid/> component Should show <GifGrid/> correctly  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Bikes
  </h3>
  <div
    className="card-grid"
  >
    <p
      className="animate__animated animate__flash"
    >
      Loading...
    </p>
  </div>
</Fragment>
`;
